lsp/src/jsonrpc.mli

```diff
@@ -135,4 +135,6 @@ end) : sig
   val notification : _ t -> Message.notification -> unit Fiber.t
 
   val request : _ t -> Message.request -> Response.t Fiber.t
+
+  val try_cancel_request : _ t -> [ `Cancelled | `Failed ] Fiber.t
```

Comment:
I'm not sure that this is the right API. We should be able to ask for a cancellation on an arbitrary request (given the Id).